docs/source/tutorials.md

```diff
@@ -101,3 +101,197 @@ Since the target repository is different than the source, it has to be specified
+useEq {True} {True} eq is = IsTrue.itsTrue
+
+reverseEq : { x : Bool } → (IsTrue x) → x ≡ True
+reverseEq {False} ()
+reverseEq {True} input = refl 
+
+proof₁ (x ∷ x₁ ∷ xs) ⦃ (ManyElem .x .x₁ .xs ⦃ h₁ ⦄ ⦃ h₂ ⦄) ⦄ = useEq ( sym $
+    begin
+        isAscending (x ∷ x₁ ∷ xs)
+    ≡⟨⟩ 
+        (if x <= x₁ then isAscending (x₁ ∷ xs) else False)
+    ≡⟨ ifTrueEqThen (x <= x₁)  (reverseEq h₂) ⟩ 
+        isAscending (x₁ ∷ xs)
+    ≡⟨ reverseEq (proof₁ (x₁ ∷ xs) ) ⟩ 
+        True
+    ∎ ) IsTrue.itsTrue
+```
+Two helper functions, useEq and reverseEq, had to be added to easily operate on the goal. Even though both of them have trivial proofs, they need to be stated explicitly so that the proper type signature can be invoked. To inspect other helper functions used in the proof, try loading the source code in agda - the definitions all come from `Haskell.Prelude`. 
+
+The reverse direction of the iff proof is, again, substantially more complicated. First, the necessary helper proofs will be discussed.
+
+```agda
+--reductio ad absurdum
+absurd₁ : {x : Bool} → (x ≡ True) → (x ≡ False) → ⊥
+absurd₁ {False} () b 
+absurd₁ {True} a ()
+```
+Reductio ad absurdum is a necessary tactic for dealing with self-contradictory statements. If such statement is one of the input arguments, this contradiction can be discarded with the [`()` keyword](https://agda.readthedocs.io/en/latest/language/function-definitions.html#absurd-patterns) in place of the contradictory argument. However, if the absurd is arising from some combination of the input arguments, it requires some helper method. 
+
+```agda
+--inductive hypothesis for isAscending function
+helper₁ : ⦃ iOrdA : Ord a ⦄ (x : a) (xs : List a) → isAscending xs ≡ False → (isAscending (x ∷ xs)) ≡ False
+helper₁ x xs h₁ with (isAscending (x ∷ xs)) in h₂
+helper₁ x (x₁ ∷ xs) h₁ | True  with (x <= x₁)
+helper₁ x (x₁ ∷ xs) h₁ | True | True = magic (absurd₁ h₂ h₁)
+helper₁ x (x₁ ∷ xs) h₁ | True | False = sym h₂
+helper₁ x xs h₁ | False = refl
+```
+Here is a helper method for the inductive hypothesis. Notice that where in the predicate syntax we were able to pattern-match on the different constructor, when working with a function, the only way to narrow down to different cases is to pattern match on the possible values of the function output. The [`with ... in` syntax](https://agda.readthedocs.io/en/latest/language/with-abstraction.html) can be used in such cases. In the first useage of the syntax, the output of the function is needed to be applied in the proof, so it needs to be saved *in* a value to be accessed in the context. This is what applying `in h₂` achieves. In the second usage, Agda manages to simplify the necessary arguments automatically, so it is not necessary to add the assertion to the syntax. 
+
+The `absurd₁` function can be applied to the `magic` function to resolve the internal contradiction. `magic` lives in `Haskell.Prim` which also needs to be imported. 
```

Comment:
It's the other way around: the `magic` function is applied to the result of the `absurd` function.